contracts/protocol/integration/auction-price/BoundedStepwiseExponentialPriceAdapter.sol

```diff
@@ -0,0 +1,198 @@
+
+        // Protect against exponential argument overflow
+        if (timeBucket > type(uint256).max / timeCoefficient) {
+            return _getBoundaryPrice(isDecreasing, maxPrice, minPrice);
+        }
+        int256 expArgument = int256(timeCoefficient * timeBucket);
+        
+        // Protect against exponential overflow and increasing relative error
+        if (expArgument > MAX_EXP_ARG) {
+            return _getBoundaryPrice(isDecreasing, maxPrice, minPrice);
+        }
+        uint256 expExpression = uint256(FixedPointMathLib.expWad(expArgument));
+
+        // Protect against priceChange overflow
+        if (scalingFactor > type(uint256).max / expExpression) {
+            return _getBoundaryPrice(isDecreasing, maxPrice, minPrice);
+        }
+        uint256 priceChange = scalingFactor * expExpression - WAD;
+
+        if (isDecreasing) {
+            // Protect against price underflow
+            if (priceChange > initialPrice) {
+                return minPrice;
+            }
+            return FixedPointMathLib.max(initialPrice - priceChange , minPrice);
+        } else {
+            // Protect against price overflow
+            if (priceChange > type(uint256).max - initialPrice) {
+                return maxPrice;
+            }
+            return FixedPointMathLib.min(initialPrice + priceChange, maxPrice);
+        }
+    }
+
+    /**
+     * @dev Returns true if the price adapter is valid for the given parameters.
+     * 
+     * @param _priceAdapterConfigData   Encoded data for configuring the price adapter.
+     * 
+     * @return isValid                  Boolean indicating if the adapter config data is valid.
+     */
+    function isPriceAdapterConfigDataValid(
+        bytes memory _priceAdapterConfigData
+    )
+        external
+        pure
+        returns (bool isValid)
+    {
+        (
+            uint256 initialPrice,
+            uint256 scalingFactor,
+            uint256 timeCoefficient,
+            uint256 bucketSize,
+            bool isDecreasing,
+            uint256 maxPrice,
+            uint256 minPrice
+        ) = getDecodedData(_priceAdapterConfigData);
+
+        return areParamsValid(initialPrice, scalingFactor, timeCoefficient, bucketSize, isDecreasing, maxPrice, minPrice);
+    }
+
+    /**
+     * @dev Returns true if the price adapter parameters are valid.
+     * 
+     * @param _initialPrice      Initial price of the auction
+     * @param _scalingFactor     Scaling factor for exponential expression
+     * @param _timeCoefficient   Scaling factor for exponential argument
+     * @param _bucketSize        Time elapsed between each bucket
+     * @param _maxPrice          Maximum price of the auction
+     * @param _minPrice          Minimum price of the auction
+     */
+    function areParamsValid(
+        uint256 _initialPrice,
+        uint256 _scalingFactor,
+        uint256 _timeCoefficient,
+        uint256 _bucketSize,
+        bool /* _isDecreasing */,
```

Comment:
This function is not part of an interface apparently. Why include an unused variable here ?

Reply:
removed here [2dd4e73](https://github.com/IndexCoop/index-protocol/pull/18/commits/2dd4e73e50951f6a7c543cfebe17e1ce84099209)